packages/angular/tests/index.test.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { NavigationEnd, NavigationStart, type Router } from '@angular/router'
+import { bugsnagBootstrapper, AngularRoutingProvider } from '../lib'
+import { MockSpanFactory } from '@bugsnag/js-performance-test-utilities'
+
+jest.useFakeTimers()
+
+describe('angular integration', () => {
+  const spanFactory = new MockSpanFactory()
+
+  describe('listenForRouteChanges', () => {
+    it('attaches a handler to router events and calls startRouteChangeSpan when fired', () => {
+      const router = {
+        events: {
+          subscribe: jest.fn()
+        }
+      } as unknown as Router
+
+      const routingProvider = new AngularRoutingProvider()
+      bugsnagBootstrapper.useFactory(router)
+
+      const span = spanFactory.startSpan(
+        '[FullPageLoad]/some-route',
+        { }
+      )
+
+      const startRouteChangeSpan = jest.fn(() => ({
+        ...spanFactory.toPublicApi(span),
+        end: jest.fn()
+      }))
+
+      routingProvider.listenForRouteChanges(startRouteChangeSpan)
+
+      jest.advanceTimersByTime(51)
+
+      expect(router.events.subscribe).toHaveBeenCalled()
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const handler = jest.mocked(router.events.subscribe).mock.calls[0][0]!
+
+      const navStart = new NavigationStart(0, 'url')
+      // eslint-disable-next-line no-useless-call
+      handler.call(undefined, navStart)
```

Comment:
does this effectively mock a navigation event?

Reply:
Its just checking things are wired up:
- that we call `router.events.subscribe` with a callback handler function
- that when that callback handler function gets called with a NavigationStart event `startRouteChangeSpan` is called
- that when that callback handler function gets called with a NavigationEnd event  the span returned by `routeChangeSpan` has its `end` method called